### ISSUE TITLE
rvm::user Installation Hangs after libffi-dev package

### ISSUE THREAD

**Issue by jcderose** [sous-chefs/chef-rvm#345]
I'm trying to install RVM for a specific user ("my-user") via test kitchen, but my `kitchen converge` process hangs after the libffi-dev package installs.

Here's my output: https://gist.github.com/jcderose/969a81e816a9372c987b

I've tried destroying and recreating the instance, I've tried restarting my local machine (where test kitchen is running). Neither of these things helped the issue. 

Here's my cookbook (where the recipe is called):

```
  include_recipe 'rvm::user'
```

And here's my attributes file:

```
default['rvm']['default_ruby'] = 'ruby-2.2.3'
default['rvm']['user_default_ruby'] = 'ruby-2.2.3'
default['rvm']['user_installs'] = [
  { 'user' => 'my-user',
    'default_ruby'  => 'ruby-2.2.3',
    'rubies'        => ['ruby-2.2.3'],
    'rvmrc'         => {
      'rvm_project_rvmrc'             => 1,
      'rvm_gemset_create_on_use_flag' => 1,
      'rvm_pretty_print_flag'         => 1
    },

    'global_gems'   => []
  }
]

default['rvm']['user_global_gems'] = [
  { 'name'    => 'bundler'  }
]
```

**Comment by wallerjake** [comment]
I am having the same issue.

**Comment by vizvamitra** [comment]
+1, v0.9.4

**Comment by vizvamitra** [comment]
**Tl;dr**: Solution for Ubuntu: Add `apt_package 'gawk'` before rvm installation

First of all, here's my config:

```
default['rvm']['user_installs'] = [
  {
    'user'          => default['user']['name'],
    'default_ruby'  => '2.3.0',
    'rubies'        => ['2.3.0'],
    'global_gems'   => [ {'name'    => 'bundler'} ]
  }
]
```

After some investigation I found that in my case chef freezes after `/home/deploy/.rvm/bin/rvm 'install' 'ruby-2.3.0'` command. This command checks ruby dependencies and if some of them are missing installs them. Package installation requires sudo password, so `rvm install ...` will ask you for it. This is when chef freezes. It waits the command to exit, but it will never do so.

**chef-rvm** installs all ruby dependencies before `rvm install` execution to prevent it asking for password, but it is currently outdated. Installation of ruby 2.3.0 (and, perhaps, 2.2.3 too) additionally requires "gawk" package.

For me adding `apt_package 'gawk'` before `include_recipe 'rvm::user'` fixed the issue

**Comment by burdandrei** [comment]
Ok,
the issue is rvm is trying to install dependencies, and one of them is `libgmp-dev` here's [commit](https://github.com/rvm/rvm/commit/fa08b3913948db7084306b4252ffa16543fc2c5e) in rvm adding it as a dependency.
Good news - it is installed by default in 16.04.

**Comment by sikajs** [comment]
+1, rvm cookbook version '0.9.4', stuck while installing libffi-dev. 
Tried **apt_package 'gawk'** only, but didn't work. 
Eventually solved by installing both of 'gawk' and 'libgmp-dev' with apt-package before **include_recipe 'rvm:user'**

**Comment by damacus** [comment]
Closing due to inactivity. 

If this is still an issue please reopen or open another issue. Alternatively drop by the #sous-chefs channel on the [Chef Community Slack](http://community-slack.chef.io/) and we'll be happy to help! 

Thanks,
Sous-Chefs
